src/Microsoft.Extensions.DependencyModel/Resolution/PackageCompilationAssemblyResolver.cs

```diff
@@ -11,74 +11,95 @@ namespace Microsoft.Extensions.DependencyModel.Resolution
     public class PackageCompilationAssemblyResolver: ICompilationAssemblyResolver
     {
         private readonly IFileSystem _fileSystem;
-        private readonly string _nugetPackageDirectory;
+        private readonly string[] _nugetPackageDirectories;
 
         public PackageCompilationAssemblyResolver()
             : this(EnvironmentWrapper.Default, FileSystemWrapper.Default)
         {
         }
 
         public PackageCompilationAssemblyResolver(string nugetPackageDirectory)
-            : this(FileSystemWrapper.Default, nugetPackageDirectory)
+            : this(FileSystemWrapper.Default, new string[] {nugetPackageDirectory})
         {
         }
 
         internal PackageCompilationAssemblyResolver(IEnvironment environment,
             IFileSystem fileSystem)
-            : this(fileSystem, GetDefaultPackageDirectory(environment))
+            : this(fileSystem, GetDefaultProbeDirectories(environment))
         {
         }
 
-        internal PackageCompilationAssemblyResolver(IFileSystem fileSystem, string nugetPackageDirectory)
+        internal PackageCompilationAssemblyResolver(IFileSystem fileSystem, string[] nugetPackageDirectories)
         {
             _fileSystem = fileSystem;
-            _nugetPackageDirectory = nugetPackageDirectory;
+            _nugetPackageDirectories = nugetPackageDirectories;
         }
 
-        private static string GetDefaultPackageDirectory(IEnvironment environment) => 
-            GetDefaultPackageDirectory(RuntimeEnvironment.OperatingSystemPlatform, environment);
+        private static string[] GetDefaultProbeDirectories(IEnvironment environment) =>
+            GetDefaultProbeDirectories(RuntimeEnvironment.OperatingSystemPlatform, environment);
 
-        internal static string GetDefaultPackageDirectory(Platform osPlatform, IEnvironment environment)
+        internal static string[] GetDefaultProbeDirectories(Platform osPlatform, IEnvironment environment)
         {
-            var packageDirectory = environment.GetEnvironmentVariable("NUGET_PACKAGES");
+#if !NETSTANDARD1_3            
+#if NETSTANDARD1_6
+            var probeDirectories = AppContext.GetData("PROBING_DIRECTORIES");
+#else
+            var probeDirectories = AppDomain.CurrentDomain.GetData("PROBING_DIRECTORIES");
+#endif
 
-            if (!string.IsNullOrEmpty(packageDirectory))
-            {
-                return packageDirectory;
-            }
+           var listOfDirectories = probeDirectories as string;
+
+           if (!string.IsNullOrEmpty(listOfDirectories))
+           {
+               return listOfDirectories.Split(new char [] {Path.PathSeparator}, StringSplitOptions.RemoveEmptyEntries );
+           }
+#endif
+
+           var packageDirectory = environment.GetEnvironmentVariable("NUGET_PACKAGES");
+
+           if (!string.IsNullOrEmpty(packageDirectory))
+           {
+               return new string[] {packageDirectory};
+           }
+
+           string basePath;
```

Comment:
A thing to always remember is that DependencyModel is used on desktop as well as .NET Core.  Also, it is used on netcoreapp1.0 and netcoreapp1.1 where the host won't be passing in the probing paths.

So we should not be changing this logic at all.  This is existing logic, which @ramarag is preserving in the case where `PROBING_DIRECTORIES` are not getting passed to the DependencyModel.